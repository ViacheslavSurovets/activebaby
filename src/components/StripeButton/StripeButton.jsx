import React from 'react';
import StripeCheckout from 'react-stripe-checkout';
import { ab } from '@assets/stripe';


const StripeButton = ( { price } ) => {
  const priceForStripe = price * 100;
  const publishableKey = 'pk_test_hUzNhA4UrDu7p7jaXvSQAXi000ackM1ZDU';

  const onToken = token => {
    console.log(token);
    alert('Payment Successful');
  };

  return (
    <StripeCheckout
      label='Pay Now'
      name='Active Baby'
      billingAddress
      shippingAddress
      image={ab}
      description={`Your total is $${price}`}
      amount={priceForStripe}
      panelLabel='Pay Now'
      token={onToken}
      stripeKey={publishableKey}
    />
  );
};

export default StripeButton;
