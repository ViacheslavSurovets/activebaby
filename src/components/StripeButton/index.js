export { default as StripeButton } from './StripeButton';
