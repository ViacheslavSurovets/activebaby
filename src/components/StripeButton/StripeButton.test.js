import React from 'react';
import { shallow } from 'enzyme';
import StripeButton from './StripeButton';


describe ( 'StripeButton', () => {

  it ( 'StripeButton Snapshot', () => {
    expect ( shallow ( <StripeButton /> ) ).toMatchSnapshot ();
  } );
} );
