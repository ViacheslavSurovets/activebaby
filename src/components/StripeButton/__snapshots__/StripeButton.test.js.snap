// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StripeButton StripeButton Snapshot 1`] = `ShallowWrapper {}`;
